lib/internal/freelist.js

```diff
@@ -9,18 +10,29 @@ class FreeList {
     this.name = name;
     this.ctor = ctor;
     this.max = max;
-    this.list = [];
+    this.list = ObjectSetPrototypeOf([], null);
   }
 
   alloc() {
-    return this.list.length > 0 ?
-      this.list.pop() :
-      ReflectApply(this.ctor, this, arguments);
+    const { list } = this;
+    const { length } = list;
```

Comment:
You don't reuse the `list` variable, let's get rid of it.
```suggestion
    const { length } = this.list;
```